### PR TITLE
fix inappropriate func name

### PR DESCRIPTION
**Pull request by wulianglongrd**

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -879,7 +879,7 @@ func (c *Controller) GetProxyServiceTargets(proxy *model.Proxy) []model.ServiceT
 	return nil
 }
 
-func (c *Controller) serviceInstancesFromWorkloadInstance(si *model.WorkloadInstance) []model.ServiceTarget {
+func (c *Controller) serviceTargetsFromWorkloadInstance(si *model.WorkloadInstance) []model.ServiceTarget {
```
Comment (wulianglongrd) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:882]: fix inappropriate names.

File: pilot/pkg/serviceregistry/serviceentry/conversion.go [hunk 2]
```diff
@@ -409,7 +409,7 @@ func convertWorkloadInstanceToServiceInstance(workloadInstance *model.WorkloadIn
 // Convenience function to convert a workloadEntry into a WorkloadInstance object encoding the endpoint (without service
 // port names) and the namespace - k8s will consume this workload instance when selecting workload entries
 func (s *Controller) convertWorkloadEntryToWorkloadInstance(cfg config.Config, clusterID cluster.ID) *model.WorkloadInstance {
-	we := ConvertWorkloadEntry(cfg)
```
Comment (wulianglongrd) [inline pilot/pkg/serviceregistry/serviceentry/conversion.go:412]: duplicate with

https://github.com/istio/istio/blob/f3f71b046f5cc79895103071a6af5f4ff4ca53d1/pilot/pkg/serviceregistry/serviceentry/controller.go#L225

File: pilot/pkg/serviceregistry/serviceentry/controller.go [hunk 3]
```diff
@@ -702,9 +702,9 @@ func (s *Controller) ResyncEDS() {
 // running this function. Queueing ensures latest updated wins.
 func (s *Controller) edsUpdate(instances []*model.ServiceInstance) {
 	// Find all keys we need to lookup
-	keys := map[instancesKey]struct{}{}
+	keys := sets.NewWithLength[instancesKey](len(instances))
```
Comment (wulianglongrd) [inline pilot/pkg/serviceregistry/serviceentry/controller.go:705]: `instances` can be very large and it makes sense to specify the length.

File: pilot/pkg/serviceregistry/serviceentry/conversion.go (hunk 2 above)
Comment (howardjohn) [reply pilot/pkg/serviceregistry/serviceentry/conversion.go:412]: This seems easy to break in the future, can we make it clear it needs to be pre-converted?

Comment (wulianglongrd) [reply pilot/pkg/serviceregistry/serviceentry/conversion.go]: I haven't thought of a good way yet, so I reversed it first.

**Pull request merged**